DAOS-14311 cart: Temporary (non-landable) change to eliminate protocol query RPCs.

jgmoore-or opened the pull request:
Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

jgmoore-or committed:
DAOS-14311 cart: Temporary (non-landable) change to eliminate protocol query RPCs.

Required-githooks: true

Signed-off-by: Joseph Moore <joseph.moore@intel.com>

daosbuild1 commented on src/object/cli_mod.c:
```diff
@@ -35,9 +37,14 @@ dc_obj_init(void)
 	dc_obj_proto_version = 0;
-	rc = daos_rpc_proto_query(obj_proto_fmt_0.cpf_base, ver_array, 2, &dc_obj_proto_version);
+	rc = daos_rpc_proto_query(obj_proto_fmt_0.cpf_base, ver_array, 2,
+                                  &dc_obj_proto_version);
```
```suggestion
				  &dc_obj_proto_version);
```

daosbuild1 replied:
```suggestion
				  &dc_obj_proto_version);
```

daosbuild1 replied:
```suggestion
				  &dc_obj_proto_version);
```

daosbuild1 replied:
```suggestion
				  &dc_obj_proto_version);
```

daosbuild1 commented on src/pool/cli.c:
```diff
@@ -38,11 +38,18 @@ int	dc_pool_proto_version;
 	dc_pool_proto_version = 0;
-	rc = daos_rpc_proto_query(pool_proto_fmt_v4.cpf_base, ver_array, 2, &dc_pool_proto_version);
+	rc = daos_rpc_proto_query(pool_proto_fmt_v4.cpf_base, ver_array, 2,
+                                  &dc_pool_proto_version);
```
```suggestion
				  &dc_pool_proto_version);
```

daosbuild1 replied:
```suggestion
				  &dc_pool_proto_version);
```

daosbuild1 replied:
```suggestion
				  &dc_pool_proto_version);
```

daosbuild1 replied:
```suggestion
				  &dc_pool_proto_version);
```

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13273/1/execution/node/149/log

jgmoore-or committed:
DAOS-14311 cart: Patch to disable protocol queury.

Required-githooks: true

Signed-off-by: Joseph Moore <joseph.moore@intel.com>

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13273/2/execution/node/148/log

jgmoore-or committed:
DAOS-14311 cart: Bypass protocol query step.

Skip-unit-tests: true
Skip-nlt: true
Skip-func-test: true

Required-githooks: true

Signed-off-by: Joseph Moore <joseph.moore@intel.com>

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13273/3/execution/node/144/log

jgmoore-or committed:
DAOS-14311 cart: Bypass protocol queries.

Skip-nlt: true

Required-githooks: true

Signed-off-by: Joseph Moore <joseph.moore@intel.com>

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13273/4/execution/node/149/log

jgmoore-or committed:
DAOS-14311 cart: Diable protocol query.

Skip-nlt: true

Required-githooks: true

Signed-off-by: Joseph Moore <joseph.moore@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium Verbs Provider completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-13273/5/testReport/
